api/src/main/java/com/okta/idx/sdk/api/wrapper/AuthenticationWrapper.java

```diff
@@ -164,72 +184,617 @@ public static AuthenticationResponse authenticate(IDXClient client, Authenticati
+
+        IDXClientContext idxClientContext = null;
+        List<FormValue> enrollProfileFormValues;
+
+        NewUserRegistrationResponse newUserRegistrationResponse = new NewUserRegistrationResponse();
+
+        try {
+            idxClientContext = client.interact();
+            Assert.notNull(idxClientContext, "IDX client context may not be null");
+
+            IDXResponse introspectResponse = client.introspect(idxClientContext);
+            String stateHandle = introspectResponse.getStateHandle();
+            Assert.hasText(stateHandle, "State handle may not be null");
+
+            RemediationOption[] remediationOptions = introspectResponse.remediation().remediationOptions();
+            printRemediationOptions(remediationOptions);
+
+            RemediationOption selectEnrollProfileRemediationOption =
+                    extractRemediationOption(remediationOptions, RemediationType.SELECT_ENROLL_PROFILE);
+
+            EnrollRequest enrollRequest = EnrollRequestBuilder.builder()
+                    .withStateHandle(stateHandle)
+                    .build();
+
+            // enroll new user
+            IDXResponse enrollResponse = selectEnrollProfileRemediationOption.proceed(client, enrollRequest);
+
+            RemediationOption[] enrollRemediationOptions = enrollResponse.remediation().remediationOptions();
+            printRemediationOptions(enrollRemediationOptions);
+
+            RemediationOption enrollProfileRemediationOption =
+                    extractRemediationOption(enrollRemediationOptions, RemediationType.ENROLL_PROFILE);
+
+            enrollProfileFormValues = Arrays.stream(enrollProfileRemediationOption.form())
+                    .filter(x-> "userProfile".equals(x.getName()))
+                    .collect(Collectors.toList());
+
+            newUserRegistrationResponse.setFormValues(enrollProfileFormValues);
+
+        } catch (ProcessingException e) {
+            Arrays.stream(e.getErrorResponse().getMessages().getValue()).forEach(msg -> newUserRegistrationResponse.addError(msg.getMessage()));
+            logger.error("Something went wrong! {}, {}", e, newUserRegistrationResponse.getErrors());
+        } catch (IllegalArgumentException e) {
+            logger.error("Exception occurred", e);
+            newUserRegistrationResponse.addError(e.getMessage());
+        }
+
+        newUserRegistrationResponse.setIdxClientContext(idxClientContext);
+        return newUserRegistrationResponse;
+    }
+
+    /**
+     * Register new user with the supplied user profile reference.
+     *
+     * @param client                the IDX Client
+     * @param idxClientContext      the IDC Client context
+     * @param userProfile           the user profile
+     * @return the Authentication response
+     */
+    public static AuthenticationResponse processRegistration(IDXClient client, IDXClientContext idxClientContext, UserProfile userProfile) {
```

Comment:
Should these method names be verbs instead? `register`?

Reply:
updated.